### PR TITLE
Fixed splitting Neighbours packet

### PR DESCRIPTION
**Pull request by arkpar**

### CODE REVIEW

File: util/src/network/discovery.rs [hunk 1]
```diff
@@ -407,25 +407,37 @@ impl Discovery {
...
 			rlp.append(&nearest[n].id);
-			if (n + 1) % limit == 0 || n == nearest.len() - 1 {
-				self.send_packet(PACKET_NEIGHBOURS, &from, &rlp.drain());
-				trace!(target: "discovery", "Sent {} Neighbours to {:?}", n, &from);
+			count -= 1;
+			if count == 0 {
+				packets.push(rlp.out());
 				rlp = RlpStream::new_list(1);
-				rlp.begin_list(cmp::min(limit, nearest.len() - n));
+				count = cmp::min(limit, nearest.len() - n);
```
Comment (tomusdrw) [inline util/src/network/discovery.rs]: Shouldn't it be: `cmp::min(limit, n)`?

File: util/src/network/discovery.rs [hunk 2]
```diff
@@ -407,25 +407,37 @@ impl Discovery {
...
+		rlp.begin_list(count);
 		for n in 0 .. nearest.len() {
 			rlp.begin_list(4);
 			nearest[n].endpoint.to_rlp(&mut rlp);
 			rlp.append(&nearest[n].id);
-			if (n + 1) % limit == 0 || n == nearest.len() - 1 {
-				self.send_packet(PACKET_NEIGHBOURS, &from, &rlp.drain());
-				trace!(target: "discovery", "Sent {} Neighbours to {:?}", n, &from);
+			count -= 1;
+			if count == 0 {
```
Comment (tomusdrw) [inline util/src/network/discovery.rs]: What do you think about moving this to the beginning of the loop and start with `count=0 && isFirst=false` flags to simplify logic with first/subsequent rlps?

File: util/src/network/discovery.rs [hunk 3]
```diff
@@ -407,25 +407,37 @@ impl Discovery {
...
+	}
+
+	fn prepare_neighbours_packets(nearest: &[NodeEntry]) -> Vec<Bytes> {
+		let mut packets = Vec::new();
 		let mut rlp = RlpStream::new_list(1);
-		rlp.begin_list(cmp::min(limit, nearest.len()));
+		let limit = (MAX_DATAGRAM_SIZE - 109) / 90;
+		let mut count = cmp::min(limit, nearest.len()); 
+		rlp.begin_list(count);
 		for n in 0 .. nearest.len() {
```
Comment (tomusdrw) [inline util/src/network/discovery.rs]: Have you considered https://doc.rust-lang.org/std/slice/struct.Chunks.html (http://doc.rust-lang.org/std/primitive.slice.html#method.chunks)?

**Comment by arkpar** [comment]
merged into `beta`

**Pull request merged**